docs/authorization.md

```diff
@@ -0,0 +1,166 @@
+
+For example, if for expression `action == Action::"viewPhoto"` we bind the `action` variable to the entity `Action::"viewPhoto"`, then the result is `true`. That's because replacing `action` with `Action::"viewPhoto"` gives expression `Action::"viewPhoto" == Action::"viewPhoto"` which is obviously `true`. 
+
+As another example, consider the expression `resource.tags.contains("Private")`. If we bind variable `resource` to the entity `Photo::"vacation94.jpg"` we get `Photo::"vacation94.jpg".tags.contains("Private")`. Evaluating further, we need to look up `Photo::"vacation94.jpg"` in our entities data, and then extract its `tags` attribute. If that attribute contains a set with the string `"Private"` in it, we'll get `true`; if it's a set without `"Private"` we'll get `false`; otherwise `tags` is either not a valid attribute or contains a non-set, so we will get `error`.
+
+### Policy satisfaction<a name="policy-satisfaction"></a>
+
+Determining whether a policy satsifies a request is a straightforward use of expression evaluation. To explain it, let's introduce some notation. For a policy _c_:
++ _Principal(c)_ is the constraint involving `principal` in _c_'s policy scope. If there is no constraint on `principal`, then _Principal(c)_ is just `true`
++ _Action(c)_ is the constraint involving `action` in _c_'s policy scope, or `true` if there is no constraint
++ _Resource(c)_ is the constraint involving `resource` in _c_'s policy scope, or `true` if there is no constraint
++ _Conds(c)_ is the list of `when` and `unless` expressions in _c_
+
+Here's how we evaluate a policy _c_ with respect to a _PARC_ request. First we test whether _c_ **matches** the request, as follows:
+1. Bind `principal` to _P_ in expression _Principal(c)_ and evaluate it
+1. Bind `action` to _A_ in expression _Action(c)_ and evaluate it
+1. Bind `resource` to _R_ in expression _Resource(c)_ and evaluate it
+
+If all three steps evaluate to `true`, then _c_ matches the request. Otherwise it does not. (As it turns out, Cedar's design ensures that none of these three steps can possibly evaluate to `error`.)
+
+If _c_ matches the request, we evaluate its conditions _Conds(c)_ in order. We bind the `principal`, `action`, `resource`, and `context` variables to the _PARC_ values when we do so. If all of the `when` conditions evaluate to `true`, and all of the `unless` conditions evaluate to `false`, then policy _c_ satsifies the request, and the final evaluation result is `true`. If evaluating any condition expression yields `error` then policy evaluation halts at that point (any remaining conditions are skipped), and `error` is returned as the final result. Otherwise, `false` is returned.
+
+## Detailed Example<a name="policy-evaluation-example"></a>
```

Comment:
I've struggled with scrolling when reading this section.  Would it make sense to re-arrange the text so that (1) the input comes first and then a list where (2) each policy is immediately followed by the description of how it evaluates on this input?  So, we'd be keeping the policy and the result close together to minimize scrolling.  (Some scrolling is still needed to remember the input though!)